client/src/main/proto/spine/base/command.proto

```diff
@@ -79,13 +79,23 @@ message CommandContext {
     // The `namespace` attribute must be defined for commands in multitenant applications.
     Namespace namespace = 5;
 
-    oneof time_opts {
-        // The delay to wait before sending the command.
-        google.protobuf.Duration delay = 6;
+    // The command scheduling options.
+    Schedule schedule = 6;
+}
+
+message Schedule {
+    oneof options {
+        // The amount of time between the current moment and a command delivering to the target.
+        google.protobuf.Duration delay = 1;
 
-        // The time when the command should be sent.
-        google.protobuf.Timestamp sending_time = 7;
+        // The time when the command should be delivered to the target.
```

Comment:
The delivery time depends on the path a command goes after it is posted to `CommandBus`. There could be one or more dispatchers in between, for example. The load time for an aggregate may be unpredictable, etc.

So, in general, we won't be able to guarantee the delivery by the time specified. It would be too complex. What we can promise is that we'd post the command to the `CommandBus` shortly after the time specified.